### ISSUE TITLE
Is the `treading.Lock` really useful?

### ISSUE THREAD

**Issue by ddanier** [team23/async-signals#1]
Currently we just copied what Django had. But I think we can remove the lock.

**Comment by skewty** [comment]
Not really; ou should be using `asyncio.Lock` instead. Yes, it is very useful and very important to prevent race conditions while updating state. Don't forget that every time you call await another async task can go change something after you already checked its value..  same race-condition issue as multi-threaded code and the reason for the lock.
